### PR TITLE
[B5] settings: API keys page

### PR DESCRIPTION
**Pull request by orangejenny**
## Product Description
This is the last view in settings.

It adds datepickers to our tempus dominus library, and it migrates the base CRUD page that's used in a few parts of HQ.

before
Screenshot 2024-05-09 at 9 38 06 PM

after
Screenshot 2024-05-09 at 9 37 35 PM

### Safety story
UI changes, risk is pretty well limited to this page.

### Automated test coverage

Not much if any at the UI level.

### QA Plan

https://dimagi.atlassian.net/browse/QA-6513

### Rollback instructions
- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Review by jingcheng16 — COMMENTED** [review]
Hi Jenny, I don't have enough knowledge to review the js changes you made in this PR. But left two questions for html changes.

File: corehq/apps/hqwebapp/templates/hqwebapp/bootstrap5/base_paginated_crud.html [hunk 1]
```diff
@@ -63,11 +62,7 @@
               <tbody>
                 <tr data-bind="visible: isPaginatedListEmpty">
                   <td colspan="{{ pagination.num_columns }}">
-                    <div id="pagination-empty-notification"
-                         class="alert alert-info"
-                         style="margin: 0;">
-                      {{ pagination.text.empty }}
-                    </div>
+                    {{ pagination.text.empty }}
```
[the remark this answers is not in the document]
Comment (orangejenny) [reply corehq/apps/hqwebapp/templates/hqwebapp/bootstrap5/base_paginated_crud.html:65]: This is an intentional, though minor, visual change. Previously, if you didn't have any items, you saw a message in a blue bubble within the table, something like this (this exact screenshot is from a different page, but it's the same UI):

Screenshot 2024-05-13 at 1 25 22 PM

Now it's a plainer message:
Screenshot 2024-05-13 at 1 25 38 PM

File: corehq/apps/hqwebapp/templates/hqwebapp/bootstrap5/base_paginated_crud.html [hunk 2]
```diff
@@ -138,22 +133,16 @@
...
-            </label>
-            <select class="form-control" id="pagination-limit" data-bind="event: {change: updateListLimit}">
-              {% for limit in pagination.limit_options %}
-                <option value="{{ limit }}">
-                  {{ limit }} {{ pagination.text.limit }}
-                </option>
-              {% endfor %}
-            </select>
-          </div>
+          <select class="form-control" id="pagination-limit" data-bind="event: {change: updateListLimit}">
```
[the remark this answers is not in the document]
Comment (orangejenny) [reply corehq/apps/hqwebapp/templates/hqwebapp/bootstrap5/base_paginated_crud.html:139]: I've been doing it consistently, yes. The label isn't really adding value (and the inline form needed to align the text plus the select box properly isn't supported in B5).

**Review by millerdev — APPROVED** [review]

**Comment by millerdev** [comment]
Aside, and not a criticism of this PR: maybe it's a sign that I'm old, but I find the new inline labels and fields layout to be harder to read than the two-column layout on a large screen. I think it was changed to accommodate mobile devices, and if that's true I think it's a place where responsive design might be nice (use columns on large screen, inline on small).

**Comment by biyeun** [comment]
@orangejenny @millerdev I think the form might be harder to follow now due to lack of whitespace. usually the fields should be wrapped in `mb-3` to add a margin at the bottom.

**Comment by orangejenny** [comment]
@biyeun Good eye. https://github.com/dimagi/commcare-hq/pull/34592/ affected crispy rendering in some way such that the `mb-3` spacing wasn't being added. Fixed in f9e382a88b273758f9050c956548e16b79314a40, and the form does now have a bit more breathing room.
Screenshot 2024-05-14 at 5 46 59 PM

**Comment by jingcheng16** [comment]
Just curious, (if the two column layout refers to left column being label and right column being input field ),  why the two column layout becomes one column in this PR? I cannot identify the related code changes...

**Comment by orangejenny** [comment]
@jingcheng16 It's almost invisible. Because the form on this page uses crispy forms, adding the `use_bootstrap5` decorator is enough to do the layout switch. Code like [this template](https://github.com/dimagi/crispy-bootstrap3to5/blob/main/crispy_bootstrap3to5/templates/bootstrap3to5/field.html) in our crispy3to5 repo checks the B5 flag to determine what kind of layout to display.

There are also places that key off the containing `form_class` CSS class, like [this](https://github.com/dimagi/crispy-bootstrap3to5/blob/f8b70d6bba1d6df7510561222ed04d430be19cbc/crispy_bootstrap3to5/templates/bootstrap3to5/bootstrap5/inputs.html#L2). Those made it necessary to make the change in https://github.com/dimagi/commcare-hq/pull/34593/commits/f9e382a88b273758f9050c956548e16b79314a40 which removes the `form-horizontal` class for B5 forms (the `crispy.py` change specifically, which overrides [this line](https://github.com/dimagi/commcare-hq/blob/f9e382a88b273758f9050c956548e16b79314a40/corehq/apps/hqwebapp/crispy.py#L24) earlier in `HQFormHelper`).

**Review by millerdev — APPROVED** [review]

**Comment by jingcheng16** [comment]
@orangejenny I see.Thank you for the explanation and all the links! ❤️

**Review by millerdev — APPROVED** [review]

**Pull request merged**